Sources/XcodesKit/RuntimeInstaller.swift

```diff
@@ -0,0 +1,182 @@
+        self.runtimeList = runtimeList
+        self.sessionService = sessionService
+    }
+
+    public func printAvailableRuntimes(includeBetas: Bool) async throws {
+        let downloadables = try await runtimeList.downloadableRuntimes(includeBetas: includeBetas)
+        var installed = try await runtimeList.installedRuntimes()
+        for (platform, downloadables) in Dictionary(grouping: downloadables, by: \.platform).sorted(\.key.order) {
+            Current.logging.log("-- \(platform.shortName) --")
+            for downloadable in downloadables {
+                let matchingInstalledRuntimes = installed.remove { $0.build == downloadable.simulatorVersion.buildUpdate }
+                let name = downloadable.visibleIdentifier
+                if !matchingInstalledRuntimes.isEmpty {
+                    for matchingInstalledRuntime in matchingInstalledRuntimes {
+                        switch matchingInstalledRuntime.kind {
+                            case .bundled:
+                                Current.logging.log(name + " (Bundled with selected Xcode)")
+                            case .diskImage, .legacyDownload:
+                                Current.logging.log(name + " (Downloaded)")
+                        }
+                    }
+                } else {
+                    Current.logging.log(name)
+                }
+            }
+        }
+        Current.logging.log("\nNote: Bundled runtimes are indicated for the currently selected Xcode, more bundled runtimes may exist in other Xcode(s)")
+    }
+
+    public func downloadAndInstallRuntime(identifier: String, to destinationDirectory: Path, with downloader: Downloader, shouldDelete: Bool) async throws {
+        let downloadables = try await runtimeList.downloadableRuntimes()
+        guard let matchedRuntime = downloadables.first(where: { $0.visibleIdentifier == identifier }) else {
+            throw Error.unavailableRuntime(identifier)
+        }
+
+        if matchedRuntime.contentType == .package && !Current.shell.isRoot() {
+            throw Error.rootNeeded
+        }
+
+        let dmgUrl = try await downloadOrUseExistingArchive(runtime: matchedRuntime, to: destinationDirectory, downloader: downloader)
+        switch matchedRuntime.contentType {
+            case .package:
+                try await installFromPackage(dmgUrl: dmgUrl, runtime: matchedRuntime)
+            case .diskImage:
+                try await installFromImage(dmgUrl: dmgUrl)
+        }
+        if shouldDelete {
+            Current.logging.log("Deleting Archive")
+            try? Current.files.removeItem(at: dmgUrl)
+        }
+    }
+
+    private func installFromImage(dmgUrl: URL)  async throws {
+        Current.logging.log("Installing Runtime")
+        try await Current.shell.installRuntimeImage(dmgUrl).asVoid().async()
+    }
+
+    private func installFromPackage(dmgUrl: URL, runtime: DownloadableRuntime) async throws {
```

Comment:
#picky - can we add a few comments in this function explaing a bit more what we are doing for a few steps?

Reply:
Done 👍🏼